src/com/owncloud/android/services/observer/InstantUploadsObserverFactory.java

```diff
@@ -0,0 +1,46 @@
+/**
+ *   ownCloud Android client application
+ *
+ *   @author David A. Velasco
+ *   Copyright (C) 2017 ownCloud GmbH.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License version 2,
+ *   as published by the Free Software Foundation.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package com.owncloud.android.services.observer;
+
+import android.content.Context;
+import android.os.Build;
+
+import com.owncloud.android.db.PreferenceManager.InstantUploadsConfiguration;
+import com.owncloud.android.lib.common.utils.Log_OC;
+
+/**
+ * Builds new instances of {@link InstantUploadsObserver}, using the appropritate implementation.
+ */
+class InstantUploadsObserverFactory {
+
+    private static final String TAG = InstantUploadsObserverFactory.class.getName();
+
+    public static InstantUploadsObserver newObserver(InstantUploadsConfiguration config, Context context) {
+        if (Build.MODEL.toLowerCase().contains("nexus") ||
```

Comment:
I guess that this is a temporal solution because it's not kind of scalable to check the different device models to use an InstantUploadsObserver or the other one. 

Besides, it is possible that people with Nexus or Pixel devices use cameras different to the Google one and find problems with instant uploads if we use **InstantUploadsObserverBasedOnINotify** instead of **InstantUploadsObserverBasedOnCommonsIOFileMonitor**.

 If I'm not wrong, it's more a matter of the camera app than of the device models.

Reply:
Sure we do - we do know why FileObserver works on some devices and doesn't on others :P You'll also notice polling doesn't work on a bunch of devices too :P But yea... scroll through all the commits first ;)